### PR TITLE
Make get_sample_data autoload npy/npz files.

### PR DESCRIPTION
**Pull request by anntzer**
This makes examples shorter and more to the point by simplifying the
data-load part.  A small (standard) backcompat-dance is needed.

Edit: will also depend on #15645.
Edit: now good to go.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/specialty_plots/topographic_hillshading.py [hunk 1]
```diff
@@ -22,23 +22,21 @@
...
-    # and *dy* are assumed to be in the same units as your input z-values.
-    # Therefore, we'll need to convert the given dx and dy from decimal degrees
-    # to meters.
-    dx, dy = dem['dx'], dem['dy']
-    dy = 111200 * dy
-    dx = 111200 * dx * np.cos(np.radians(dem['ymin']))
-    #-------------------------------------------------------------------------
+dem = get_sample_data('jacksboro_fault_dem.npz', np_load=True)
+z = dem['elevation']
+#-- Optional dx and dy for accurate vertical exaggeration ----------------
```
Comment (timhoffm) [inline examples/specialty_plots/topographic_hillshading.py]:
```suggestion
#-- Optional dx and dy for accurate vertical exaggeration --------------------
```

File: examples/specialty_plots/topographic_hillshading.py [hunk 2]
```diff
@@ -22,23 +22,21 @@
...
+# of the grid (i.e. the *dx* and *dy* parameters).  Otherwise, any *vert_exag*
+# value you specify will be relative to the grid spacing of your input data
+# (in other words, *dx* and *dy* default to 1.0, and *vert_exag* is calculated
+# relative to those parameters).  Similarly, *dx* and *dy* are assumed to be in
+# the same units as your input z-values.  Therefore, we'll need to convert the
+# given dx and dy from decimal degrees to meters.
+dx, dy = dem['dx'], dem['dy']
+dy = 111200 * dy
+dx = 111200 * dx * np.cos(np.radians(dem['ymin']))
+#-------------------------------------------------------------------------
```
Comment (timhoffm) [inline examples/specialty_plots/topographic_hillshading.py]:
```suggestion
#-----------------------------------------------------------------------------
```

File: lib/matplotlib/cbook/__init__.py [hunk 3]
```diff
@@ -469,7 +469,7 @@ def is_scalar_or_string(val):
     return isinstance(val, str) or not np.iterable(val)
 
 
-def get_sample_data(fname, asfileobj=True):
+def get_sample_data(fname, asfileobj=True, *, np_load=False):
```
Comment (timhoffm) [inline lib/matplotlib/cbook/__init__.py]: `np_load` should be mentioned in the docstring.

File: examples/specialty_plots/topographic_hillshading.py (hunk 1 above)
Comment (anntzer) [reply examples/specialty_plots/topographic_hillshading.py]: fixed

File: examples/specialty_plots/topographic_hillshading.py (hunk 2 above)
Comment (anntzer) [reply examples/specialty_plots/topographic_hillshading.py]: fixed

File: lib/matplotlib/cbook/__init__.py (hunk 3 above)
Comment (anntzer) [reply lib/matplotlib/cbook/__init__.py]: fixed

**Comment by anntzer** [comment]
rebased

**Comment by anntzer** [comment]
Well the whole point *is* to change the default to True in the future; if you don't want to do that I'm not even sure the PR is worth it.
How many people actually want to have the path to our sample data, as opposed to loading our sample data? (Well sure technically we don't know, however I think we can make a reasonable guess...)

**Comment by anntzer** [comment]
rebased

**Pull request merged**